.github/workflows/ci.yml

```diff
@@ -9,33 +10,308 @@ on:
       - main
 
 jobs:
-  test:
-    if: ${{ github.repository == 'open-telemetry/opentelemetry-ruby' }}
+  base:
     strategy:
+      fail-fast: false
       matrix:
-        include:
-          - name: Test Linux / Ruby 2.7
-            os: ubuntu-latest
-            ruby: "2.7"
-            flags: --include-simple --include-appraisal
-          - name: Test Linux / Ruby 3.0
-            os: ubuntu-latest
-            ruby: "3.0"
-            flags: --include-simple --include-appraisal
-          - name: Test Linux / Ruby 3.1
-            os: ubuntu-latest
-            ruby: "3.1"
-            flags: --include-simple --include-appraisal
-          - name: Test Linux / JRuby
-            os: ubuntu-latest
-            ruby: jruby
-            # otlp omitted because it depends on protobuf which does not work on JRuby
-            flags: --include-simple --exclude opentelemetry-exporter-otlp
+        gem:
+          - opentelemetry-api
+          - opentelemetry-common
+          - opentelemetry-metrics-api
+          - opentelemetry-registry
+          - opentelemetry-resource_detectors
+          - opentelemetry-sdk
+          - opentelemetry-semantic_conventions
+          - opentelemetry-test-helpers
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+    name: ${{ matrix.gem }} / ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: "Test Ruby 3.1"
+        uses: ./.github/actions/test_gem
```

Comment:
🎉 Why hasn't this ever occurred to me to do?

Reply:
Honestly I didn't think much about doing it until @veraticus showed me how to do so at `$PREV_JOB`, so all credit goes to him.

More practically, it's a useful tool but limited, so it's not always useful (in the same way that re-using workflows via `workflow_call` is also useful, but limited). It worked out pretty well here though!